src/zenml/stack_stores/base_stack_store.py

```diff
@@ -151,6 +155,37 @@ def register_stack_component(
                 and name already exists.
         """
 
+    @abstractmethod
+    def update_stack_component(
+        self,
+        component: StackComponentWrapper,
+    ) -> None:
+        """Update a stack component.
+
+        Args:
+            component: The new component to update with.
+
+        Raises:
+            KeyError: If no stack component exists with the given name.
+        """
+
+    @abstractmethod
+    def rename_stack_component(
+        self,
+        old_name: str,
+        component: StackComponentWrapper,
+    ) -> None:
+        """Rename a stack component."""
```

Comment:
To me, the signature of this method looks a little strange, but I am probably not aware of all the implications: why do I need to pass a `StackComponentWrapper` instead of just giving the `old_name` and `new_name` ?